### ISSUE TITLE
RHEL yum issue

### ISSUE THREAD

**Issue by BurningTheSky** [garethr/garethr-docker#314]
Hi, I fully assume that I may be doing something wrong but when I try to apply a docker manifest I get the following issue, could you help me with that I have done wrong?

Manifest:
include 'docker'

docker::image { 'ubuntu':
  image_tag => 'trusty',
}

Error as follows:
Error: Execution of '/bin/yum -d 0 -e 0 -y --enablerepo=rhel7-extras list docker' returned 1: Error: No matching Packages to list
Error: /Stage[main]/Docker::Install/Package[docker]/ensure: change from absent to present failed: Execution of '/bin/yum -d 0 -e 0 -y --enablerepo=rhel7-extras list docker' returned 1: Error: No matching Packages to list

**Comment by garethr** [comment]
What version of RHEL are you currently using?

The current state of docker packaging for RHEL based systems is currently best described as confusing.

The problem you are likely running into is one of repositories. You can probably get this to work with some combination of `package_name` and `repo_opt` for the moment. It's possible the package name in the repos your pointing at isn't docker but something else. Maybe try and yum search docker?

Docker Inc are now shipping supported packages as of a few weeks ago, and I'm in the process of adding those in here: https://github.com/garethr/garethr-docker/pull/311. This should make the getting started experience easier on RHEL based systems.

**Comment by BurningTheSky** [comment]
Thanks for the quick response. I am currently using Red Hat Enterprise Linux Server release 7.1 (Maipo). You are right with the package_name, when I do a yum search it shows up "docker-engine.x86_64 : The open-source application container engine" and that is in the rhel7-extras repo. I've now updated the package_name for RHEL and it looks to be working! Thanks. I did wonder if that issue would be related.

**Comment by snufalufagus** [comment]
+1 for needing the CentOS/RHEL package name and repo fix. Module isn't presently usable.

**Comment by jovandeginste** [comment]
For a workaround, see: #281

If you don't use hiera, you can try the following snippet (including 'migration' part) - you still need to change the hiera-references to actual values then:

```puppet
  $docker_version = hiera('docker::version')
  $docker_package_name = hiera('docker::package_name')

  if $docker_version {
    $dockerpackage = "${docker_package_name}-${docker_version}"
  } else {
    $dockerpackage = $docker_package_name
  }

  class {
    '::docker':
      ensure                      => present,
      manage_package              => false,
      use_upstream_package_source => false,
      ...
  }

  package { 'old-docker':
    ensure => absent,
    name   => 'docker', # or docker-io, depending on your case
  } ->
  package { $docker_package_name:
    ensure          => present,
    name            => $dockerpackage,
    install_options => '--enablerepo=some-repo',
    before          => Service['docker'],
  }
```

**Comment by jovandeginste** [comment]
Apparently, (at least in CentOS) the package is back as 'docker':

# yum info docker
<...>
Name        : docker
Arch        : x86_64
Version     : 1.7.1
Release     : 108.el7.centos
Size        : 30 M
Repo        : installed
From repo   : centos-extras

FWIW, this release integrates with FirewallD (which I was really waiting for)

**Comment by garethr** [comment]
I've tested this with the PR in #311 which now defaults to use the upstream packages for RHEL based distros. You can disable this using `use_upstream_package_source` as noted in the updated README. This should be merged soon and released in 5.0.
